### ISSUE TITLE
Disappearing service?

### ISSUE THREAD

**Issue by pikeas** [coreos/fleet#1081]
I have a three node cluster, and my_service@.service. This service has `Conflicts=my_service@*` set, and is currently running on all three nodes:

    core@node1 ~ $ fleetctl list-units
    UNIT			MACHINE				ACTIVE	SUB
    my_service@1.service	<ID1>.../<IP1>	       active	running
    my_service@2.service	<ID2>.../<IP2>	       active	running
    my_servicer@3.service	<ID3>.../<IP3>	       active	running

Now, I bounce one of my nodes:

    core@node2 ~ $ sudo reboot

    core@node1 ~ $ fleetctl list-units
    UNIT			MACHINE				ACTIVE	SUB
    my_service@1.service	<ID1>.../<IP1>	       active	running
    my_service@2.service	<ID2>.../<IP2>	       active	running

The service cannot be rescheduled to a different node, because the other two nodes have a conflict. I would expect to see my_service@3 marked in an error state. Instead, it disappears from list-units, and reappears when the other node comes back up.

Is this a bug or intentional behavior? I was expecting something like:

    my_service@3.service	                    inactive	schedule-conflict

**Comment by bcwaldon** [comment]
Use `list-unit-files` to enumerate the desired state of the cluster. `list-units` represents just the "live" state

**Comment by pikeas** [comment]
@bcwaldon This seems like an bug in list-units. At the very least, it's counterintuitive and should be documented.

I schedule three instances S1, S2, S3. One of them dies and cannot be rescheduled. I still scheduled three units, so three units should be listed, and one of them should be in a failure state.

**Comment by bcwaldon** [comment]
@pikeas What does the output of `list-unit-files` look like after this machine failure happens?

**Comment by robszumski** [comment]
@bcwaldon I agree with @pikeas that could be a better experience. What do you think of adding a line at the end of the list-units like `3 units can't be scheduled`:

```
$ fleetctl list-units
UNIT				MACHINE				ACTIVE	SUB
registry.1.service		fe2f2cdf.../10.28.63.xxx	active	running
registry.2.service		4e5fdea7.../10.182.158.xxx	active	running
registry.3.service		8316b1eb.../10.28.176.xxx	active	running

3 units can't be scheduled. View with list-unit-files
```

**Comment by pikeas** [comment]
@bcwaldon The output in `list-unit-files` is correct. However, based on the names of the commands, `list-unit-files` is information about *files* I've submitted, and `list-units` is information about units I've scheduled to run. When I want to know the state of my system, I check `list-units`, not `list-unit-files`.

@robszumski Your suggestion is a step in the right direction, but I believe it needs to go further. A unit that cannot be scheduled should be treated as (and I would say is by definition) a failed unit, which should receive an appropriate failure flag in `list-units`.
